modevo-script/src/main/java/giis/modevo/migration/script/execution/ScriptExecution.java

```diff
@@ -43,57 +43,94 @@ public void execute(Script script, CassandraConnection c, String nameKeyspace) {
 				Row rw = resultIt.next();
 				List<ColumnValue> cvs = getColumnValuesSearchRow(rw, s);
 				List<Select> selectsInside = highFor.getSelectsInsideFor();
-				executeSelects (cvs, selectsInside, c, nameKeyspace);
-				executeInserts (script, cvs, highFor, c, nameKeyspace);			
+				List<List<ColumnValue>> cvsFromWhere = executeSelects (cvs, selectsInside, c, nameKeyspace);
+				executeInserts (script, cvs, cvsFromWhere, highFor, c, nameKeyspace);			
 			}
 		}
 	}
 	/**
 	 * Execution of the insert statements inside a FOR loop
+	 * @param cvsFromWhere 
 	 */
-	private void executeInserts(Script script, List<ColumnValue> cvs, For highFor, CassandraConnection c, String nameKeyspace) {
+	private void executeInserts(Script script, List<ColumnValue> cvs, List<List<ColumnValue>> cvsFromWhere, For highFor, CassandraConnection c, String nameKeyspace) {
 		for (Insert i: script.getInserts()) {
 			if (i.getInsideFor().equals(highFor)){
 				String insertStatement = i.getInsertStatement();
-				for (ColumnValue cv : i.getColumnValue()) {
-					String variableName = cv.getVariableName();
-					if (variableName == null) {
-						variableName = cv.getColumnSelectOrigin().getVariableName();
-					}
-					ColumnValue cvValues = findCVNameVariable (cvs, variableName);
-					if (cvValues == null) {
-						insertStatement= insertStatement.replace(variableName, "''");
-						continue;
-					}
-					insertStatement= insertStatement.replace(variableName, "'"+cvValues.getValue()+"'");
-				}
 				String nameTableInsert = i.getNameTable();
 				String statementInsertWithKeyspace = insertStatement.replace("INSERT INTO "+nameTableInsert, "INSERT INTO "+"\""+nameKeyspace+"\"."+nameTableInsert);
-				c.executeStatement(statementInsertWithKeyspace);
+				List<String> insertsInside = new ArrayList<>();
+				for (ColumnValue cv : cvs) {
+					statementInsertWithKeyspace=replaceVariableName (statementInsertWithKeyspace, cv, cvs);
+				}
+				for (List<ColumnValue> listColumnValues : cvsFromWhere) {
+					List<String> replacedVariableNamesList = replaceVariableNamesListValues (listColumnValues, statementInsertWithKeyspace);
+					insertsInside.addAll(replacedVariableNamesList);
+				}
+				if (cvsFromWhere.isEmpty()) {
+					String statementWithEmptyValues = statementInsertWithKeyspace.replaceAll("\\$\\d+", "''");
+					insertsInside.add(statementWithEmptyValues);	
+				}
+				for (String statementToExecute : insertsInside) {
+					c.executeStatement(statementToExecute);
+				}
 			}
 		}	
 	}
+	/**
+	 * Creates a list of statements with its values replaced by the values included in the list.
+	 */
+	private List<String> replaceVariableNamesListValues(List<ColumnValue> listColumnValues,
+			String statementWithVariables) {
+		List<String> listStatements = new ArrayList<>();
+		String statementReplacedValues = statementWithVariables;
+		boolean replaced = false;
```

Comment:
this flag is not necessary, return from inside loop when found

Reply:
Removed the flag, although with a different solution as there could be more than one replacement in the loop.